_posts/2020-10-27-solidity-0.8.x-preview.md

```diff
@@ -0,0 +1,192 @@
+
+We did not introduce checks for explicit type conversions from larger to smaller
+types, because we think that such checks might be unexpected. We would love to get
+your input on that matter!
+
+The following operations now have checks that they did not have before:
+ - addition (``+``), subtraction (``-``), multiplication (``*``)
+ - increment and decrement (``++`` / ``--``)
+ - unary negation (``-``)
+ - exponentiation (``**``)
+ - division (see below) (``/``)
+
+There are some edge cases you might not expect.
+For example, unary negation is only available for signed types.
+The problem is that in two's complement representation, there is one more negative number for each bit width
+than there are positive numbers. This means that the folloing code will trigger a Panic:
+
+```solidity
+int8 x = -128;
+-x;
+```
+
+The reason is that ``128`` does not fit an 8-bit signed integer.
+
+For the same reason, signed division can result in an assertion:
+```solidity
+int8 x = -128;
+x/(-1);
+```
+
+Everyone who wanted to write a ``SafeMath`` function for exponentiation probably noticed
+that it is rather expensive to do that because the EVM does not provide overflow signalling.
+Essentially you have to implement your own ``exp`` routine without using the ``exp`` opcode
+for the general case.
+
+We hope that we found a rather efficient implementation and would also appreciate your feedback about that!
+
+For many special cases, we actually implemented it using the ``exp`` opcode instead of our own implementation.
+More specifically, exponentiation operations that use a literal number as base will use the ``exp`` opcode directly
+and operations that use a base and an exponent that are both "small": There are special
+implementations for all bases up to 2, for bases up to 10 and up to 306.
+
+#### ``unchecked`` Blocks
+
+Since checked arithmetic uses more gas, as discussed in the previous section, and
+because there are times where you really want the wrapping behaviour, for example when
+implementing cryptographic routines, we provide a way to disable the checked arithmetic and
+re-enable the previous "wrapping" or "modulo" arithmetic:
+
+Any block of the form ``unchecked { ... }`` uses wrapping arithmetic. You can use
+this special block as a regular statement inside another block:
+
+```solidity
+contract C {
+    function f() public pure returns (uint) {
+        uint x = 0;
+        unchecked { x--; }
+        return x;
+    }
+}
+```
+
+An unchecked block can of course contain any number or any kind of other statements.
+If there is a function call inside the unchecked block, the function will
+not inherit the setting - you have to use another ``unchecked`` block inside the function
+if you also want it to use wrapping arithmetic.
+
+We decided on some restritions for the ``unchecked`` block:
+
+1. You cannot use ``_;`` inside it (for modifiers) because this might lead to confusion about whether the property is inherited.
```

Comment:
```suggestion
1. In modifiers you cannot use ``_;`` inside it, because this might lead to confusion about whether the property is inherited.
```

Reply:
It is weird however that we allow function calls within `unchecked` and tell people it is not inherited, but we disallow `_` for the same reason.

Reply:
Because `_` can be seen as "insert the function body here" instead of "now call the function itself".

Reply:
I know, but still.